File: src/main/java/loop/Receipt.java
package loop;

import java.util.Scanner;

public class Receipt {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int total = scanner.nextInt();
        int number = scanner.nextInt();

        int sum = 0;
        for (int i = 1; i <= number; i++) {
            int a = scanner.nextInt();
            int b = scanner.nextInt();
            sum += a * b;
        }

        scanner.close();

        if (total == sum) {
            System.out.println("Yes");
        } else {
            System.out.println("No");
        }

    }

}

File: src/main/kotlin/loop/Receipt.kt
package loop

import java.util.Scanner

fun main() {

    val scanner = Scanner(System.`in`)
    val total: Int = scanner.nextInt()
    val number: Int = scanner.nextInt()

    var sum = 0
    for (i in 1..number) {
        val a: Int = scanner.nextInt()
        val b: Int = scanner.nextInt()
        sum += a * b
    }

    scanner.close()

    if (total == sum) {
        println("Yes")
    } else {
        println("No")
    }

}